proposals/0000-overloaded-record-fields.rst

```diff
@@ -0,0 +1,359 @@
+.. proposal-number:: Leave blank. This will be filled in when the proposal is
+                     accepted.
+
+.. trac-ticket:: Leave blank. This will eventually be filled with the Trac
+                 ticket number which will track the progress of the
+                 implementation of the feature.
+
+.. implemented:: Leave blank. This will be filled in with the first GHC version which
+                 implements the described feature.
+
+Overloaded Record Fields
+========================
+
+This is a proposal to introduce a new extension,
+``OverloadedRecordFields``, to make it easier to work with record
+datatypes that reuse the same field names.  It also makes some changes
+to the existing ``OverloadedLabels`` extension for consistency.
+
+
+Motivation
+----------
+
+A serious limitation of the Haskell record system is the inability to
+overload field names in record types: for example, if the data types
+
+::
+
+  data Person  = Person  { personId :: Int, name :: String }
+  data Address = Address { personId :: Int, address :: String }
+
+are in scope in the same module, there is no way to determine which
+type an occurrence of the ``personId`` record selector refers to.  A
+common workaround is to use a unique prefix for each record type, but
+this leads to less clear code and obfuscates relationships between
+fields of different records.  Qualified names can be used to
+distinguish record selectors from different modules, but using one
+module per record is often impractical.
```

Comment:
[Yuras comments on reddit](https://www.reddit.com/r/haskell/comments/518d8q/ghc_proposal_for_overloadedrecordfields_plus_some/d7a34pg?context=3) that this section could do with further justification.